File: epfl-hc/hc2012/graders/takeawayLib2.cpp
/////////////////////////////
// DO NOT SUBMIT THIS FILE //
/////////////////////////////

#include <stdlib.h>
#include <stdio.h>
#include "takeawayLib2.hpp"
using namespace std;

#define MAX_N 50
#define UNDEF -1

int p[MAX_N], n[MAX_N], t[MAX_N];
int N, T;
int ourMin, ourMax;
int yourMin, yourMax;
int cnt;
FILE* fin;

void read_case() {
	fscanf(fin, "%d",&N);
	for (int i=0; i<MAX_N; i++)
		p[i] = n[i] = t[i] = UNDEF;
	for (int i=0; i<N; i++)
		fscanf(fin, "%d %d %d", &p[i], &n[i], &t[i]);
	fscanf(fin, "%d %d", &ourMin, &ourMax);
	yourMin = yourMax = UNDEF;
	cnt = 0;
}

void min_balloons(int yourMinBound) {
	yourMin = yourMinBound;
	cnt++;
}

void max_balloons(int yourMaxBound) {
	yourMax = yourMaxBound;
	cnt++;
}

int main() {
        fin = stdin;
	fscanf(fin, "%d",&T);
	while (T--)	{
		read_case();
		count_balloons(N,p,n,t);
		if (cnt!=2 || ourMin!=yourMin || ourMax!=yourMax) {
			printf("Incorrect\n");
			exit(0);
		}
		
	}
	printf("Correctf52cac7981d3b301410e7cab97a3c2a5\n");
	return 0;
}

File: epfl-hc/hc2012/graders/TakeawayLib2.java
/////////////////////////////
// DO NOT SUBMIT THIS FILE //
/////////////////////////////

public class TakeawayLib2 {
 private static final int MAX_N = 50;
 private static final int UNDEF  = -1;

  private static int p[];
  private static int n[];
  private static int t[];
  private static int N, T;
  private static int ourMin, ourMax;
  private static int yourMin, yourMax;
  private static int cnt;
  private static java.util.Scanner scanner;

  private static void read_case() {
          N = scanner.nextInt();
          p = new int[N];
          n = new int[N];
          t = new int[N];
          for (int i=0; i<N; i++) {
            p[i] = scanner.nextInt();
            n[i] = scanner.nextInt();
            t[i] = scanner.nextInt();
          }
          ourMin = scanner.nextInt();
          ourMax = scanner.nextInt();
          yourMin = UNDEF;
          yourMax = UNDEF;
          cnt = 0;
  }

  public static void min_balloons(int yourMinBound) {
          yourMin = yourMinBound;
          cnt++;
  }

  public static void max_balloons(int yourMaxBound) {
          yourMax = yourMaxBound;
          cnt++;
  }

  public static void main(String args[]) {
    try {
      		Takeaway2 t2 = new Takeaway2();
          scanner = new java.util.Scanner(System.in);
          T = scanner.nextInt();
          while (T-- != 0)	{
                  read_case();
                  t2.count_balloons(N,p,n,t);
                  if (cnt!=2 || ourMin!=yourMin || ourMax!=yourMax) {
                          System.out.printf("Incorrect\n");
                          System.exit(0);
                  }
                  
          }
          System.out.printf("Correctf52cac7981d3b301410e7cab97a3c2a5\n");
    } catch(Throwable t) {
      System.out.println(t);
    }
  }
}
